CalPatRec/src/AgnosticHelixFinder_module.cc

```diff
@@ -640,15 +642,19 @@ namespace mu2e {
     // do isolation, average, and eDepFlag flagging
     if (_doIsolationFlag == true || _doAverageFlag == true || _doEDepFlag == true) {
       for (size_t i = 0; i < _tcHits.size(); i++) {
-        if (_tcHits[i].inHelix == true || _tcHits[i].hitIndice < 0) { continue; }
+        if (_tcHits[i].inHelix == true) { continue; }
+        if (_tcHits[i].hitIndice == HitType::STOPPINGTARGET) { continue; }
+        if (_tcHits[i].hitIndice == HitType::CALOCLUSTER) { continue; }
         int hitIndice = _tcHits[i].hitIndice;
         float hitEnergy = _chColl->at(hitIndice).energyDep();
         if (_doEDepFlag == true && hitEnergy > _eDepFlagThresh) { _tcHits[i].highEDep = true; }
         if (_doIsolationFlag == true || _doAverageFlag == true) {
           int nHitsNear = 0;
           XYZVectorF seedPos = getPos(i);
           for (size_t j = 0; j < _tcHits.size(); j++) {
-            if (_tcHits[j].inHelix == true || _tcHits[j].hitIndice < 0) { continue; }
+            if (_tcHits[j].inHelix == true) { continue; }
```

Comment:
Please restructure this logic:
- consolidate the clauses into a single if()
- reverse the logic to avoid use of 'continue'
- put the fastest test first (i ==j)